File: src/dataset/python/change_exercism-ThomasZumsteg/claude/ClaudeSonnet4_change_v1.py
def find_minimum_coins(total_change, coins):
    if total_change == 0:
        return []
    
    # Sort coins in descending order for better pruning
    sorted_coins = sorted(coins, reverse=True)
    
    # Use dynamic programming with memoization
    memo = {}
    
    def dp(remaining):
        if remaining == 0:
            return []
        if remaining < 0:
            return None
        if remaining in memo:
            return memo[remaining]
        
        best = None
        for coin in sorted_coins:
            if coin <= remaining:
                sub_result = dp(remaining - coin)
                if sub_result is not None:
                    candidate = [coin] + sub_result
                    if best is None or len(candidate) < len(best):
                        best = candidate
        
        memo[remaining] = best
        return best
    
    result = dp(total_change)
    if result is None:
        raise ValueError("Cannot make chage for {} from {}".format(
            total_change, coins))
    return result

def fewer(possibly_none, other):
    if possibly_none is None:
        return False 
    return len(possibly_none) < len(other)

File: src/dataset/javascript/change_exercism-ThomasZumsteg/claude/ClaudeSonnet4_change_v1.js
class Change {
    calculate(coins, value) {
        if(value < 0)
            throw new Error("Negative totals are not allowed.");
        
        if(value === 0) return [];
        
        const sortedCoins = coins.sort((a, b) => b - a);
        const memo = new Map();
        
        const findMinChange = (remaining) => {
            if(remaining === 0) return [];
            if(remaining < 0) return null;
            if(memo.has(remaining)) return memo.get(remaining);
            
            let minChange = null;
            
            for(const coin of sortedCoins) {
                if(coin > remaining) continue;
                
                const subResult = findMinChange(remaining - coin);
                if(subResult !== null) {
                    const currentChange = [coin, ...subResult];
                    if(minChange === null || currentChange.length < minChange.length) {
                        minChange = currentChange;
                    }
                }
            }
            
            memo.set(remaining, minChange);
            return minChange;
        };
        
        const result = findMinChange(value);
        if(result === null)
            throw new Error("The total " + value + " cannot be represented in the given currency.");
        
        return result.sort((a, b) => a - b);
    }
}

module.exports = Change;